### PR TITLE
Fix `--suggest-typed` errors

### LINKED ISSUE #1951: srb init is marking files as false/ignore when it shouldn't
Running `srb init` in a project where files already have `typed` sigils attached at values higher than `false` will sometimes cause these files to be downgraded to `typed: false` or `typed: ignore`. It shouldn't be downgrading these files, especially if they have been checked successfully in the past.

## Related issues
#1945

### LINKED ISSUE #2426: suggest-typed marks files using T.all as false
#### Input

This is related to #1951.

We start with this file, which comes from [the Intersection Types doc](https://sorbet.org/docs/intersection-types):

```ruby
# typed: true
extend T::Sig

module I1
  def i1; end
end

module I2
  def i2; end
end

class C
  include I1
  include I2
end

class D
  include I1
  include I2
end

sig { params(x: T.all(I1, I2)).void }
def foo(x)
  x.i1
  x.i2
end

foo(C.new)
foo(D.new)
```

It passes typechecking just fine, and I have no reason to believe it should be invalid.

The reason I believe the issue is coming from `T.all` specifically is because suggest-typed marks the file as `true` if you replace that line with `sig { params(x: T.untyped).void }`.

#### Observed output

After running suggest-typed, it becomes `typed: false`.

#### Expected behavior

I'd expect it to keep the file marked as `typed: true`.

I've created a repository for demonstrating this issue: https://github.com/connorshea/sorbet-t-all-suggest-typed-issue

In [the initial commit](https://github.com/connorshea/sorbet-t-all-suggest-typed-issue/commit/4033c1e7eecb5467eadc27c0884da52c8e2d5b8d), I init Sorbet and add the `foo.rb` file. Then in [the next commit](https://github.com/connorshea/sorbet-t-all-suggest-typed-issue/commit/247c7e47f326038c5a45c3afc0a9be73e0320468) I run `bundle exec srb rbi suggest-typed`, which causes `foo.rb` to become `typed: false`.

You can clone it and see the problem by running these commands:

```sh
git clone https://github.com/connorshea/sorbet-t-all-suggest-typed-issue
git checkout 4033c1e7eecb5467eadc27c0884da52c8e2d5b8d
bundle install
cat foo.rb #=> you'll see that it's `typed: true`
bundle exec srb tc #=> No errors! Great job.
bundle exec srb rbi suggest-typed
cat foo.rb #=> you'll see that it's now `typed: false`
```

### Other notes

I wanted to note here that this issue has been impacting the controllers in my Rails app since September, and I only just now went through the trouble of trying to determine the source of the issue by bisecting, which took a solid 45 minutes (and that's after I tried to do a bisect of the sorbet versions to see when the regression was introduced a few months ago).

It'd be great if suggest-typed had test coverage for issues like this to prevent them from happening in the future, because they're _incredibly_ annoying to track down.

### PR DESCRIPTION
**Pull request by paracycle**
There are two problems with the way `--suggest-typed` is working:

1. `MultipleBehaviourDefs` error seems to be an internal Stripe-only error that is raised for when a type has behaviour coming from multiple files.

	Under normal circumstances, that error is never visible for end-users unless `--stripe-mode` flag is enabled. However, the error is still being considered for setting the minimum error level of each file, and is thus breaking the behaviour of `--suggest-typed`. As a result many files that are passing type checking with strictness levels `false` or above are being suggested `ignore` instead.

	This PR fixes the problem by ignoring `MultipleBehaviorDefs` when setting the minimum error level of a file.

2. `T.all` check for existence of methods in all the components of the `AndType`. However, that means that an `UnknownMethod` error is being raised each time one of the method existence checks fails for one of the components in a `T.all`, even though the method would be found in another component. This, again, means that files that pass type checking with strictness levels `true` or above are now being suggested `false` instead.

	This PR fixes the problem by making sure the error gets properly queued in the result from the method existence check so that the call can properly register the final error, if it exists, when it has to.

### Motivation

Fixes #1951
Fixes #2426

### Test plan

See included automated tests.

### CODE REVIEW

File: core/GlobalState.cc [hunk 1]
```diff
@@ -1415,7 +1415,8 @@ void GlobalState::_error(unique_ptr<Error> error) const {
     }
     auto loc = error->loc;
     if (loc.file().exists() && error->what != errors::Infer::SuggestTyped &&
-        error->what != core::errors::Resolver::SigInFileWithoutSigil) {
+        error->what != core::errors::Resolver::SigInFileWithoutSigil &&
+        error->what != core::errors::Namer::MultipleBehaviorDefs) {
```
Comment (DarkDimius) [inline core/GlobalState.cc]: Can we check for stripe mode here too to make it work correctly in both modes?

Comment (paracycle) [reply core/GlobalState.cc]: Ah sure, but I am not too sure how to pass the `stripeMode` flag into the `GlobalState` object. Let me look around.

Comment (paracycle) [reply core/GlobalState.cc]: Ok, I moved the definition of which error types should be ignored out of the `GlobalState` class. This should be cleaner now.

Comment (paracycle) [reply core/GlobalState.cc]: I am open to better naming on the newly added methods/fields.

File: core/types/calls.cc [hunk 2]
```diff
@@ -499,7 +499,13 @@ DispatchResult dispatchCallSymbol(const GlobalState &gs, DispatchArgs args,
             return DispatchResult(Types::untypedUntracked(), std::move(args.selfType), Symbols::untyped());
         }
         auto result = DispatchResult(Types::untypedUntracked(), std::move(args.selfType), Symbols::noSymbol());
-        if (auto e = gs.beginError(args.locs.call, errors::Infer::UnknownMethod)) {
+        // We want to always be able to build the error object
```
Comment (DarkDimius) [inline core/types/calls.cc]:
```suggestion
        // This is a hack. We want to always be able to build the error object
```

Comment (DarkDimius) [reply core/types/calls.cc]: Note that the same applies to argument adaptation, right?
Happy to have it done either in this PR or another one

Comment (paracycle) [reply core/types/calls.cc]: Hmm, I am not too familiar with the argument adaptation part. Would love to take a look if you show me the way, though.

**Comment by paracycle** [comment]
@DarkDimius this should be good to merge. I'd happy to handle the argument adaptation part in another PR.

**Comment by wpride** [comment]
This is huge! Thank you.

**Pull request merged**